src/compiler/utilities.ts

```diff
@@ -254,6 +254,25 @@ namespace ts {
         return !nodeIsMissing(node);
     }
 
+    /**
+     * Appends a range of value to begin of an array, returning the array.
+     *
+     * @param to The array to which `value` is to be appended. If `to` is `undefined`, a new array
+     * is created if `value` was appended.
+     * @param from The values to append to the array. If `from` is `undefined`, nothing is
+     * appended. If an element of `from` is `undefined`, that element is not appended.
+     */
+    export function prependStatements<T extends Statement>(to: T[], from: ReadonlyArray<T> | undefined): T[] | undefined {
+        if (from === undefined || from.length === 0) return to;
+        if (to === undefined) return from.slice();
+        const prologue = to.length && isPrologueDirective(to[0]) && to.shift();

```

Comment:
It's possible that there is more than one directive. Though I don't know if that occurs in real world code. Should this case be handled here?

Reply:
We don't handle it in any other locations we check for prologues, I looked. I figured I shouldn't differ here.